From 0a1b0fd929a8bbdf7c47b418b8d0ee6a8de3a7a3 Mon Sep 17 00:00:00 2001
Message-Id: <0a1b0fd929a8bbdf7c47b418b8d0ee6a8de3a7a3.1649663728.git.geliang.tang@suse.com>
From: Jiapeng Chong <jiapeng.chong@linux.alibaba.com>
Date: Mon, 8 Feb 2021 17:43:36 +0800
Subject: [PATCH] bpf: Simplify bool comparison
Git-commit: 0a1b0fd929a8bbdf7c47b418b8d0ee6a8de3a7a3
Patch-mainline: v5.12-rc2
References: bsn#127
Modified-by-SEL: No


Fix the following coccicheck warning:

./tools/bpf/bpf_dbg.c:893:32-36: WARNING: Comparison to bool.

Reported-by: Abaci Robot <abaci@linux.alibaba.com>
Signed-off-by: Jiapeng Chong <jiapeng.chong@linux.alibaba.com>
Signed-off-by: Andrii Nakryiko <andrii@kernel.org>
Link: https://lore.kernel.org/bpf/1612777416-34339-1-git-send-email-jiapeng.chong@linux.alibaba.com
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 tools/bpf/bpf_dbg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/bpf/bpf_dbg.c b/tools/bpf/bpf_dbg.c
index a0ebcdf59c31..a07dfc479270 100644
--- a/tools/bpf/bpf_dbg.c
+++ b/tools/bpf/bpf_dbg.c
@@ -890,7 +890,7 @@ static int bpf_run_stepping(struct sock_filter *f, uint16_t bpf_len,
 	bool stop = false;
 	int i = 1;
 
-	while (bpf_curr.Rs == false && stop == false) {
+	while (!bpf_curr.Rs && !stop) {
 		bpf_safe_regs();
 
 		if (i++ == next)
-- 
2.34.1

